src/main/java/com/github/fishio/audio/AudioEngine.java

```diff
@@ -0,0 +1,416 @@
+package com.github.fishio.audio;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.value.ChangeListener;
+
+import javax.sound.sampled.LineUnavailableException;
+import javax.sound.sampled.UnsupportedAudioFileException;
+
+import com.github.fishio.logging.Log;
+import com.github.fishio.logging.LogLevel;
+
+/**
+ * Singleton class for managing playing audio.
+ */
+public final class AudioEngine {
+	/**
+	 * The path to the directory that contains the music.
+	 */
+	private static final String DIR_MUSIC = "/sound/music/steel_drum_island/";
+	
+	/**
+	 * The maximum amount of threads that can play sounds concurrently.
+	 */
+	private static final int MAX_THREAD_COUNT = 11;
```

Comment:
See the requirements: "The audio engine shall support playing at least 10 sounds at the same time". Since the background music is always playing, 11 - 1 = 10.

<layout